yt_dlp/extractor/youtube.py

```diff
@@ -3307,7 +3307,57 @@ def _extract_heatmap(self, data):
                 'value': ('intensityScoreNormalized', {float_or_none}),
             })) or None
 
-    def _extract_comment(self, comment_renderer, parent=None):
+    def _extract_comment(self, view_model, entities, parent=None):
+        comment_entity_payload = get_first(entities, ('payload', 'commentEntityPayload', {dict}))
+        toolbar_entity_payload = get_first(entities, ('payload', 'engagementToolbarStateEntityPayload', {dict})) or {}
+        comment_id = traverse_obj(comment_entity_payload, ('properties', 'commentId', {str}))
+        if not comment_id:
+            return
+
+        info = {
+            'id': comment_id,
+            'parent': parent or 'root',
+            **traverse_obj(comment_entity_payload, {
+                'text': ('properties', 'content', 'content', {str}),
+                'like_count': ('toolbar', 'likeCountA11y', {parse_count}),
+                'author_id': ('author', 'channelId', {self.ucid_or_none}),
+                'author': ('author', 'displayName', {str}),
+                'author_thumbnail': ('author', 'avatarThumbnailUrl', {url_or_none}),
+            }),
+        }
+
+        # Timestamp is an estimate calculated from the current time and time_text
+        time_text = traverse_obj(comment_entity_payload, ('properties', 'publishedTime', {str})) or ''
+        timestamp = self._parse_time_text(time_text)
+
+        info.update({
+            # FIXME: non-standard, but we need a way of showing that it is an estimate.
+            '_time_text': time_text,
+            'timestamp': timestamp,
+        })
+
+        info['author_url'] = urljoin(
+            'https://www.youtube.com',
+            traverse_obj(comment_entity_payload, ('author', 'channelCommand', 'innertubeCommand', (
+                ('browseEndpoint', 'canonicalBaseUrl'), ('commandMetadata', 'webCommandMetadata', 'url'))),
+                expected_type=str, get_all=False))
+
+        author_is_uploader = traverse_obj(comment_entity_payload, ('author', 'isCreator', {bool}))
+        if author_is_uploader is not None:
+            info['author_is_uploader'] = author_is_uploader
+
+        if toolbar_entity_payload.get('heartState') == 'TOOLBAR_HEART_STATE_HEARTED':
+            info['is_favorited'] = True
+
+        if traverse_obj(comment_entity_payload, ('author', 'isVerified', {bool})):
+            info['author_is_verified'] = True
+
+        if traverse_obj(view_model, 'pinnedText'):
+            info['is_pinned'] = True
+
+        return info
```

Comment:
```suggestion
        comment_entity_payload = get_first(entities, ('payload', 'commentEntityPayload', {dict}))
        if not (comment_id := traverse_obj(comment_entity_payload, ('properties', 'commentId', {str}))):
            return

        toolbar_entity_payload = get_first(entities, ('payload', 'engagementToolbarStateEntityPayload', {dict}))
        time_text = traverse_obj(comment_entity_payload, ('properties', 'publishedTime', {str})) or ''

        return {
            'id': comment_id,
            'parent': parent or 'root',
            **traverse_obj(comment_entity_payload, {
                'text': ('properties', 'content', 'content', {str}),
                'like_count': ('toolbar', 'likeCountA11y', {parse_count}),
                'author_id': ('author', 'channelId', {self.ucid_or_none}),
                'author': ('author', 'displayName', {str}),
                'author_thumbnail': ('author', 'avatarThumbnailUrl', {url_or_none}),
                'author_is_uploader': ('author', 'isCreator', {bool}),
                'author_is_verified': ('author', 'isVerified', {bool}),
                'author_url': ('author', 'channelCommand', 'innertubeCommand', (
                    ('browseEndpoint', 'canonicalBaseUrl'), ('commandMetadata', 'webCommandMetadata', 'url')
                ), {lambda x: urljoin('https://www.youtube.com', x)}),
            }, get_all=False),
            'is_favorited': (None if toolbar_entity_payload is None else  # TODO(before merge): Is this logic correct?
                             toolbar_entity_payload.get('heartState') == 'TOOLBAR_HEART_STATE_HEARTED'),
            'is_pinned': view_model.get('pinnedText'),
            '_time_text': time_text,  # FIXME: non-standard, but we need a way of showing that it is an estimate.
            'timestamp': self._parse_time_text(time_text),            
        }
```

Reply:
per comments above, do we need this?
```py
                'like_count': ('toolbar', 'likeCountA11y', {parse_count}, {lambda x: x or 0}),
```

Reply:
Can you explain? `parse_count` returns `0` correctly:
```
>>> parse_count('0')
    0
```

Reply:
I was basing this off the feedback from shoxie007, who implied that the current code omits `like_count` instead of returning a `0` value

Reply:
> ```py
>             'is_pinned': view_model.get('pinnedText'),
> ```
someone should confirm this is always a boolean value in the youtube response, or else `bool()`/`bool_or_none()` this

Reply:
> someone should confirm this is always a boolean value in the youtube response, or else `bool()`/`bool_or_none()` this

You are right. Even if it's always bool, we should do the type check anyway. Updated my suggestion to
```diff
-            'is_pinned': view_model.get('pinnedText'),
+            'is_pinned': traverse_obj(view_model, ('pinnedText', {bool})),
```

On a side note, `view_model` is being passed in here only for this one value. It may be better to compute this in the parent function instead.

Reply:
> You are right. Even if it's always bool, we should do the type check anyway. Updated my suggestion to
> ```diff
> -            'is_pinned': view_model.get('pinnedText'),
> +            'is_pinned': traverse_obj(view_model, ('pinnedText', {bool})),
> ```

the value of `view_model.get('pinnedText')` is either `None` or `'Pinned by [UPLOADER]'`

And yeah I think the proper thing to do would be to compute this in the parent function